### PR TITLE
HYPERFLEET-1291 - feat: add domain-specific custom CEL functions

### PR DESCRIPTION
**Pull request by rafabene**
## Summary

- Add 5 domain-specific CEL helper functions to reduce adapter config verbosity:
  - `conditionStatus(conditions, type)` — replaces the double-filter pattern (~12x per ROSA config)
  - `conditionAge(conditions, type)` — replaces `timestamp(now()) - timestamp(...)` chains
  - `stableFor(conditions, type, seconds)` — combines conditionStatus + conditionAge into a single stability-window check
  - `statusFeedbackValue(statusFeedback, name)` — replaces 4-step filter chain for Maestro ManifestWork feedback
  - `triState(trueCond, falseCond)` — replaces nested ternaries for "True"/"False"/"Unknown" mapping
- Updates `docs/conventions/cel.md` with signatures, descriptions, and before/after examples
- Updates `docs/adapter-authoring-guide.md` Appendix A (CEL Quick Reference) and adds simplified version of time-based stability precondition pattern
- Updates CEL showcase dry-run with patterns 21–24 demonstrating the new functions
- Bumps `.bingo/golangci-lint.mod` to `go 1.26.0` to align with [HYPERFLEET-1309](https://redhat.atlassian.net/browse/HYPERFLEET-1309)

## Test plan

- [x] 20 unit tests covering all 5 functions: happy path, absent condition, empty list, edge cases, and function composition
- [x] CEL showcase dry-run updated with 4 new patterns (conditionAge, conditionStatus, stableFor, triState)
- [x] `make test` — all 15 packages pass
- [x] `make lint` — 0 issues

[HYPERFLEET-1309]: https://redhat.atlassian.net/browse/HYPERFLEET-1309?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/adapter-authoring-guide.md [hunk 1]
```diff
@@ -1587,11 +1607,26 @@ status.conditions.filter(c, c.type == "Reconciled")
...
 status.conditions.exists(c, c.type == "Reconciled" && c.status == "True")
 
-# Get first matching element with fallback
+# Get first matching element with fallback (verbose)
 status.conditions.filter(c, c.type == "Reconciled").size() > 0
   ? status.conditions.filter(c, c.type == "Reconciled")[0].status
   : "False"
 
+# Same, using domain-specific helper (preferred)
+conditionStatus(conditions, "Reconciled")
```
[the remark this answers is not in the document]
Comment (rafabene) [reply docs/adapter-authoring-guide.md:1692]: Fixed — changed fallback from `"False"` to `"Unknown"` to match `conditionStatus()` contract.

File: docs/adapter-authoring-guide.md [hunk 2]
```diff
@@ -1641,6 +1680,10 @@ resources.?myResource.?status.?conditions.orValue([])
 <summary>Check ManifestWork statusFeedback for a namespace phase</summary>
 
 ```cel
+# Using statusFeedbackValue() helper (preferred):
+statusFeedbackValue(resources.namespace0.statusFeedback, "phase")
+
+# Equivalent verbose form:
```
[the remark this answers is not in the document]
Comment (rafabene) [reply docs/adapter-authoring-guide.md:1764]: Fixed — added `has()` guards before the helper call to prevent hard dereference when `namespace0` is absent.

File: docs/conventions/cel.md [hunk 3]
```diff
@@ -13,10 +13,20 @@ Extracted params are injected as **top-level** names — write `clusterID`, not
...
+
 - `now()` — current time as RFC3339 string
 - `toJson(val)` — serialize any value to JSON string
 - `dig(map, "dot.path")` — safe nested map access, returns null if missing
 
+### Domain-Specific
+
+- `conditionStatus(conditions, type)` — returns the `status` field (`"True"`, `"False"`, etc.) of the first condition matching `type`, or `"Unknown"` if absent
+- `conditionAge(conditions, type)` — returns elapsed seconds since `last_transition_time` for the matching condition, or `-1` if absent
+- `stableFor(conditions, type, seconds)` — returns `true` only when `conditionStatus` is `"True"` AND `conditionAge` exceeds the threshold
```
[the remark this answers is not in the document]
Comment (rafabene) [reply docs/conventions/cel.md]: Fixed — updated wording from "exceeds" to "at least" to match the `>=` implementation.

File: internal/criteria/cel_evaluator_test.go [hunk 4]
```diff
@@ -372,6 +372,187 @@ func TestCELEvaluatorCustomFunctions(t *testing.T) {
...
+
+	t.Run("triState with CEL expressions replacing nested ternaries", func(t *testing.T) {
+		result, err := evaluator.EvaluateSafe(
+			`triState(conditionStatus(conditions, "Reconciled") == "True", ` +
+				`conditionStatus(conditions, "Reconciled") == "False")`)
+		require.NoError(t, err)
+		require.False(t, result.HasError())
+		assert.Equal(t, "True", result.Value)
+	})
+}
```
[the remark this answers is not in the document]
Comment (rafabene) [reply internal/criteria/cel_evaluator_test.go:554]: Fixed — added `TestCELEvaluatorDomainFunctions_MalformedInputs` with 7 tests covering: invalid timestamp, missing `last_transition_time`, non-list `values`, non-map entry, and missing `fieldValue.string`.

File: internal/criteria/cel_evaluator.go [hunk 5]
```diff
@@ -139,7 +139,178 @@ func customCELFunctions() []cel.EnvOption {
...
+					}
+					transitionTime, ok := cond["last_transition_time"].(string)
+					if !ok {
+						return types.Bool(false)
+					}
+					t, err := time.Parse(time.RFC3339, transitionTime)
+					if err != nil {
+						return types.Bool(false)
+					}
+					return types.Bool(int64(time.Since(t).Seconds()) >= threshold)
```
[the remark this answers is not in the document]
Comment (rafabene) [reply internal/criteria/cel_evaluator.go:224]: Kept `>=` (at least) — updated the docs to match. The inclusive boundary is more intuitive: `stableFor(conditions, "Reconciled", 300)` means "stable for at least 300 seconds".

File: test/testdata/dryrun/cel-showcase/dryrun-cel-showcase-task-config.yaml [hunk 6]
```diff
@@ -432,6 +440,46 @@ post:
...
+              conditionAge(conditions, "Reconciled")
+
+          # ---------------------------------------------------------
+          # Pattern 23: stableFor() — stability window check
+          # Before: conditionStatus == "True" && conditionAge > 300
+          # (as a single combined expression with all the filter chains)
+          # ---------------------------------------------------------
+          is_reconciled_stable:
+            expression: |
+              stableFor(conditions, "Provisioned", 60)
```
[the remark this answers is not in the document]
Comment (rafabene) [reply test/testdata/dryrun/cel-showcase/dryrun-cel-showcase-task-config.yaml]: Fixed — changed from `"Provisioned"` to `"Reconciled"` to match the pattern name.
